docs/guides/es2k/es2k-guide.md

```diff
@@ -0,0 +1,212 @@
+
+#### Set environment variables
+
+- export DEPEND_INSTALL=`absolute path for installing dependencies`
+- export SDE_INSTALL=`absolute path for p4 sde install built in previous step`
+
+```bash
+source ./scripts/es2k/setup_env.sh $IPDK_RECIPE $SDE_INSTALL $DEPEND_INSTALL 
+```
+
+#### Compile the recipe
+
+```bash
+cd $IPDK_RECIPE
+./make-all.sh --target=es2k
+```
+
+*Note*: By default, `make-all.sh` will create an `install` directory under the
+networking recipe. User can specify a different directory using `--prefix`
+option to make-all.sh. The following examples assume default `install`
+directory for the executables. If not, user will need to specify the
+appropriate path instead of ./install.
+
+### Run Infrap4d
+
+#### Set up the environment required by infrap4d
+
+*Note*: `sudo` option is required when running copy_config_files.sh since
+we are creating directories and copying config file at system level.
+
+```bash
+sudo ./scripts/es2k/copy_config_files.sh $IPDK_RECIPE $SDE_INSTALL
+```
+
+#### Set hugepages required for ES2K
+
+Run the hugepages script.
+
+```bash
+sudo ./scripts/es2k/set_hugepages.sh
+```
+
+#### Export all environment variables to sudo user
+
+```bash
+alias sudo='sudo PATH="$PATH" HOME="$HOME" LD_LIBRARY_PATH="$LD_LIBRARY_PATH" SDE_INSTALL="$SDE_INSTALL"'
+```
+
+#### Modify the conf file
+
+Update `/usr/share/stratum/es2k/es2k_skip_p4.conf` with CPF BDF as per your setup.
+User can get CPF BDF by running `lspci | grep 1453`
+Note: max vport you can pass here from [0-6] "eal-args": "--lcores=1-2 -a <cpf_bdf>,vport=[0-1] -- -i --rxq=1 --txq=1 --hairpinq=1 --hairpin-mode=0x0"
+
+Eg. "eal-args": "--lcores=1-2 -a 00:01.6,vport=[0-1] -- -i --rxq=1 --txq=1 --hairpinq=1 --hairpin-mode=0x0"
+
+We give options to each process to request numbers of configure queues before starting the process
+
+Admin must set cfgqs-idx between 0-15. See the latest es2k_skip_p4.conf for the exact syntax.
+If you want 16 cfgqs, use "cfgqs-idx": "0-15". Specify the range as a string. Index numbers from 0 to 15 are supported.
+In multi process environment, user should plan and split the queues between primary and secondary processes and specify
+the range with cfgqs-idx parameter. Total number of queues split between processes should not exceed 16.
+
+#### Run the infrap4d daemon
+
+```bash
+cd $IPDK_RECIPE
+sudo ./install/sbin/infrap4d
+```
+
+*Note*: By default, infrap4d runs in detached mode. If you want to run in
+attached mode, specify the --nodetach command-line option.
+
+- All infrap4d logs are by default logged under `/var/log/stratum`.
+- All P4SDE logs are logged in `p4_driver.log` under `$IPDK_RECIPE`.
+*-All OVS logs are logged under `/tmp/ovs-vswitchd.log`.
+
+### Run a sample program
+
+Open a new terminal to set the pipeline and try the sample P4 program.
+Set up the environment and export all environment variables to sudo user.
+
+```bash
+source ./scripts/es2k/setup_env.sh $IPDK_RECIPE $SDE_INSTALL $DEPEND_INSTALL
```

Comment:
Since we are mentioning it is a new terminal, better to add a command "cd $IPDK_RECIPE" as well.

Reply:
I am creating new PR to have scripts included as part of install tree. This section of guide will be modified again once those changes are merged and tested